tests/BaseTest.php

```diff
@@ -110,19 +112,41 @@ public function tryToGetAnAccessToken(Google_Client $client)
   {
     $this->checkClientCredentials();
 
-    $client->setRedirectUri("urn:ietf:wg:oauth:2.0:oob");
-    $client->setConfig('access_type', 'offline');
     $authUrl = $client->createAuthUrl();
 
-    echo "\nPlease enter the auth code:\n";
-    ob_flush();
+    $log = tempnam(sys_get_temp_dir(), 'php_auth_log');
+    $cmd = sprintf(
+      'OAUTH_LOG_FILE=%s php -S localhost:8000 -t %s > /dev/null & echo $!',
+      $log,
+      __DIR__
+    );
+    $pid = exec($cmd);
     `open '$authUrl'`;
-    $authCode = trim(fgets(STDIN));
 
-    if ($accessToken = $client->fetchAccessTokenWithAuthCode($authCode)) {
-      if (isset($accessToken['access_token'])) {
-        return $accessToken;
+    $time = time();
+
+    // break after waiting five minutes
+    while (time() - $time < 300) {
+      if ($file = file_get_contents($log)) {
+        break;
       }
+
+      sleep(1);
+    }
+
+    if ('1' === $file) {
+      $this->markTestSkipped('Failed to retrieve access token');
+    }
+
+    if (!$accessToken = json_decode($file, true)) {
+      $this->markTestSkipped('invalid JSON: ' . $file);
+    }
+
+    // kill the web server
+    `kill $pid`;
```

Comment:
If you hit Ctrl+C while waiting for the token, what happens with the webserver process?